### PR TITLE
Assert Rearrange speed benchmarks have a minimum value

### PR DESCRIPTION
**Pull request by vincentpierre**
## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: scripts/hab2_bench/assert_bench.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+MINIMUM_PERFORMANCE_1_PROCESS = {
+    "idle_all": 280,
+    "idle_noconcur": 260,
+    "idle_nosleep": 120,
+    # "idle_render": "[Idle Render Only]",
+    "idle_single_camera_all": 510,
+    "idle_single_camera_noconcur": 480,
+    "idle_single_camera_nosleep": 150,
+    # "idle_single_camera_render": "[Idle (head-RGB) Render Only]",
+    "interact_all": 290,
```
Comment (vincentpierre) [inline scripts/hab2_bench/assert_bench.py:30]:
```suggestion
    "interact_all": 260,
```

**Comment by Skylion007** [comment]
I know we previously looked into this library: https://pypi.org/project/pytest-benchmark/, but we never went forward with getting it fully setup.

File: scripts/hab2_bench/assert_bench.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        assert len(run_fps) > 0, f"Missing run for {fname}"
+        mean_sps = sum(run_fps) / len(run_fps)
+        if mean_sps < minimum_performance_map[name]:
+            failed_runs.append(
+                f"The performance of the setting {name} : {name_map.get(name, 'n/a')} has a SPS regression. Was expecting a minimum SPS of {minimum_performance_map[name]} but got an average of {mean_sps}"
+            )
+
+    return failed_runs
+
+
```
Comment (Skylion007) [inline scripts/hab2_bench/assert_bench.py:82]: Add if __name__ == '__main__' so we can import the benchmark_sps func from another file if we need to without running the script.

File: scripts/hab2_bench/assert_bench.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    "idle_single_camera_noconcur": 1600,
+    "idle_single_camera_nosleep": 650,
+    # "idle_single_camera_render": "[Idle (head-RGB) Render Only]",
+    "interact_all": 920,
+    "interact_noconcur": 950,
+    "interact_nosleep": 500,
+}
+
+
+class RegressionError(Exception):
```
Comment (Skylion007) [inline scripts/hab2_bench/assert_bench.py:50]: Maybe Subclass AssertionError instead?

File: scripts/hab2_bench/assert_bench.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    "idle_single_camera_nosleep": "[Idle (head-RGB) No Auto-sleep]",
+    # "idle_single_camera_render": "[Idle (head-RGB) Render Only]",
+    "interact_all": "[Interact]",
+    "interact_noconcur": "[Interact No Concurrent Rendering]",
+    "interact_nosleep": "[Interact No Auto-sleep]",
+}
+
+MINIMUM_PERFORMANCE_1_PROCESS = {
+    "idle_all": 280,
+    "idle_noconcur": 260,
```
Comment (vincentpierre) [inline scripts/hab2_bench/assert_bench.py:23]:
```suggestion
    "idle_noconcur": 220,
```

**Review by aclegg3 — APPROVED** [review]
Thanks for adding regression testing. :+1: 
I think we can merge this for now and follow-up with other kinds of more-detailed and modular bench-marking moving forward.

Minor comment: we should add some comment strings here to identify where these golden numbers are coming from (e.g. the CI machine specs) and if there is already a regression what our target numbers would be to fix that.

File: scripts/hab2_bench/assert_bench.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    "interact_nosleep": 120,
+}
+
+MINIMUM_PERFORMANCE_16_PROCESS = {
+    "idle_all": 920,
+    "idle_noconcur": 950,
+    "idle_nosleep": 500,
+    # "idle_render": "[Idle Render Only]",
+    "idle_single_camera_all": 1450,
+    "idle_single_camera_noconcur": 1600,
```
Comment (vincentpierre) [inline scripts/hab2_bench/assert_bench.py:47]:
```suggestion
    "idle_single_camera_noconcur": 1550,
```

**Pull request merged**